src/main/java/com/icthh/xm/ms/balance/web/rest/BalanceResource.java

```diff
@@ -103,6 +104,30 @@ public ResponseEntity<BalanceDTO> updateBalance(@Valid @RequestBody BalanceDTO b
             .body(result);
     }
 
+    /**
+     * PUT  /balances : Updates an existing balance status.
+     *
+     * @param id the id of the balance to update
+     * @param status the status of the balance to update
+     * @param context the context Map
+     * @return the ResponseEntity with status 200 (OK) and with body the updated balanceDTO,
+     * or with status 400 (Bad Request) if the status is not valid,
+     * or with status 404 (Not Found) if balance not found by id
+     * or with status 500 (Internal Server Error) if the balance couldn't be updated
+     */
+    @PreAuthorize("hasPermission({'id': #id, 'status': #status, 'context': #context}, 'balance', 'BALANCE.STATUS')")
+    @PutMapping("/balances/{id}/statuses/{status}")
+    @Timed
+    @PrivilegeDescription("Privilege to updates an existing balance")
+    public ResponseEntity<BalanceDTO> updateBalanceStatus(@PathVariable Long id,
+                                                          @PathVariable String status,
+                                                          @RequestBody(required = false) Map<String, Object> context){
```

Comment:
pleas reformat all the changes, I've noticed few places with missing spaces

Reply:
fixed